maven-core/src/main/java/org/apache/maven/lifecycle/internal/builder/BuilderCommon.java

```diff
@@ -80,15 +83,14 @@
     @Inject
     private ExecutionEventCatapult eventCatapult;
 
-    @Inject
-    private Logger logger;
-
     public BuilderCommon()
     {
+        this.logger = LoggerFactory.getLogger( getClass() );
```

Comment:
Why here?

Reply:
The ctor is needed for container, look below the 2nd ctor